Sources/SwiftKafka/Configuration/KafkaConfiguration+Security.swift

```diff
@@ -0,0 +1,453 @@
+
+extension KafkaConfiguration {
+    // MARK: - SSLConfiguration
+
+    /// Use to configure an SSL connection.
+    public struct SSLConfiguration: Sendable, Hashable {
+        /// A key either located in a file or directly as a key String (PEM format).
+        public struct Key: Sendable, Hashable {
+            internal enum _Key: Sendable, Hashable {
+                case file(location: String)
+                case pem(String)
+            }
+
+            let _internal: _Key
+
+            /// A key located in a file at the given `location`.
+            public static func file(location: String) -> Key {
+                return Key(
+                    _internal: .file(location: location)
+                )
+            }
+
+            /// A key String (PEM format).
+            public static func pem(_ pem: String) -> Key {
+                return Key(
+                    _internal: .pem(pem)
+                )
+            }
+        }
+
+        /// A SSL private key.
+        public struct PrivateKey: Sendable, Hashable {
+            /// The private key itself.
+            public var key: Key
+            /// The password associated with the private key.
+            public var password: String
+
+            public init(location: Key, password: String) {
+                self.key = location
+                self.password = password
+            }
+        }
+
+        /// A SSL key store (PKCS#12).
+        public struct KeyStore: Sendable, Hashable {
+            /// Path to the key store.
+            public var location: String
+            /// The key store's password.
+            public var password: String
+
+            public init(location: String, password: String) {
+                self.location = location
+                self.password = password
+            }
+        }
+
+        internal enum _SSLConfiguration: Sendable, Hashable {
+            case keyPair(
+                privateKey: PrivateKey,
+                publicKeyCertificate: Key,
+                caCertificate: Key,
+                crlLocation: String?
+            )
+            case keyStore(
+                keyStore: KeyStore,
+                caCertificate: Key,
+                crlLocation: String?
+            )
+        }
+
+        let _internal: _SSLConfiguration
+
+        /// Use SSL with a given private/public key pair.
+        ///
+        /// - Parameters:
+        ///
+        ///     - privateKey: The client's private key (PEM) used for authentication.
+        ///     - publicKeyCertificate: The client's public key (PEM) used for authentication.
+        ///     - caCertificate: File or directory path to CA certificate(s) for verifying the broker's key.
+        ///     - crlocation: Path to CRL for verifying broker's certificate validity.
+        public static func keyPair(
+            privateKey: PrivateKey,
+            publicKeyCertificate: Key,
+            caCertificate: Key,
```

Comment:
we might want to use different types here so we can provide a reasonable default for `caCertificate`.